test/08_manylinuxXXXX_only/spam.c

```diff
@@ -12,7 +10,9 @@ spam_system(PyObject *self, PyObject *args)
     if (!PyArg_ParseTuple(args, "s", &command))
         return NULL;
 
-#if defined(__linux__)
+#if defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 17)
+    sts = (int)secure_getenv("NON_EXISTING_ENV_VARIABLE");
+#elif defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 10)
     sts = malloc_info(0, stdout);
 #endif
```

Comment:
Sorry, I might be a little ignorant here - what does this section do? Maybe some comments would help :)

Reply:
Part of the `manylinux` standard is a maximum version of glibc from which symbols can be used. So to force a wheel to be e.g. `manylinux2010`-compatible but not `manylinux1`-compatible, we need to use a symbol from `glibc` that's in `glibc` [version 2.12](https://www.python.org/dev/peps/pep-0571/#the-manylinux2010-policy) but not in [version 2.5](https://www.python.org/dev/peps/pep-0513/#the-manylinux1-policy).

Reply:
Oh I see. A comment to that effect would be great.

Reply:
@mayeut The one thing we're actually not testing this way is if during compilation we're building in the right `manylinux` image, since it now doesn't compile things that would fail if `glibc`'s version is smaller? I know we're testing it indirectly, when checking the wheels, but ...

What about having `CIBW_ENVIRONMENT='CFLAGS="-DMANYLINUX_IMAGE={}"'.format(manylinux_image)` and using that for the precprocessor statements? That would solve both problems, if it's not too messy?

Reply:
This one feels a bit messy. I'd rather rely on `auditwheel` and checking wheels doing their job.
The simple comments added in `spam.c` should be enough don't you think ?

Reply:
Works as well, for me.

Reply:
@joerick, good for you ?

Reply:
I added a little more explanation as comment, but otherwise i'm happy :)